### PR TITLE
Create Specter Desktop tray app

### PR DESCRIPTION
**Pull request by ben-kaufman**
This will create a `.app` which can then be wrapped into dmg and give us simple installation (at least on Mac, didn't try Linux/ Windows yet).
The problem for now is that, for some unclear reason, the app works only when launched from the terminal (ie. `open ./path-to-app/Specter.app`, but doesn't work when launched by double click from Finder. Still investigating that so keeping this PR as a draft for now.

### CODE REVIEW

**Comment by stepansnigirev** [comment]
I think we should run specter server not as `--daemon`, but as a thread of this app. Whenever you quit the app the server should quit as well.
This might also fix problems with launching the app.

**Comment by stepansnigirev** [comment]
Would it make sense to start the server and open the web browser as soon as the server is ready at launch?

**Comment by stepansnigirev** [comment]
Oh, and maybe we should add "open HWI bridge settings" in that menu!

**Comment by ben-kaufman** [comment]
I was thinking to add app settings, and there allow the user to select if he uses local Specter or remote, so the remote option will switch to HWIBridge. Also the user will be able to specify URL in the remote option whih will be opened instead of the localhost, and will automatically be whitelisted for HWIBridge. What do you think?

**Comment by stepansnigirev** [comment]
Yes, I think that's a good idea. Then starting server in the beginning doesn't make sense - the user has to specify the mode first.

**Comment by ben-kaufman** [comment]
I added a Preference popup, which will automatically launch on first open to set Local/ remote, then it will open whatever the user chose automatically.
The only thing now left I think is to auto whitelist in HWIBridge the domain given in the settings when using a remote Specter.

**Comment by stepansnigirev** [comment]
Maybe binaries should be removed from the PR and added to `.gitignore`?

File: pyinstaller/specter_desktop.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+            return
+
+
+def run_specterd(menu):
+    global specterd_thread, wait_for_specterd_process
+    try:
+        specterd_command = [resource_path('specterd/{}'.format(
+            'hwibridge' if settings.value(
+                "remote_mode", defaultValue=False, type=bool
+            ) else 'specterd'
```
Comment (stepansnigirev) [inline pyinstaller/specter_desktop.py:148]: should add `.exe` extension on windows

File: pyinstaller/specter_desktop.spec [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          debug=False,
+          bootloader_ignore_signals=False,
+          strip=False,
+          upx=True,
+          upx_exclude=[],
+          runtime_tmpdir=None,
+          console=True )
+
+if sys.platform == 'darwin':
+   app = BUNDLE(exe,
```
Comment (stepansnigirev) [inline pyinstaller/specter_desktop.spec:64]: weird spacing...

File: pyinstaller/specter_desktop.spec [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                bundle_identifier=None,
+                info_plist={
+                    'NSPrincipleClass': 'NSApplication',
+                    'NSAppleScriptEnabled': False,
+                    'NSHighResolutionCapable': 'True',
+                    'NSRequiresAquaSystemAppearance': 'True',
+                    'LSUIElement': 1
+                })
+if sys.platform == 'win32' or sys.platform == 'win64' or sys.platform == 'linux':
+  exe = EXE(pyz,
```
Comment (stepansnigirev) [inline pyinstaller/specter_desktop.spec:76]: here as well

File: .gitignore [hunk 4]
```diff
@@ -10,3 +10,4 @@ dist
 .vscode
 cert.pem
 key.pem
+*.dmg
```
Comment (stepansnigirev) [inline .gitignore:13]: maybe add `pyinstaller/specterd/*` here as well?

**Pull request merged**